server/index.js

```diff
@@ -1,25 +1,75 @@
 import Hapi from '@hapi/hapi';
 import fetch from 'node-fetch'
+import pg from 'pg'; 
+import { v4 as uuidv4 } from 'uuid';
+
+const pool = new pg.Pool()
 
 const init = async () => {
     const server = Hapi.server({
         port: 3500,
-        host: 'localhost'
+        host: 'localhost',
+        routes: { cors: true },
     });
 
     server.route({
       method: 'GET',
-      path: '/diff',
-      options: {
-        cors: true,
-      },
+      path: '/github-diff',
       handler: async (request, h) => {
           // FIXME add validation to only fetch diffs, e.g. it has to end with .diff
           const response = await fetch(request.query.url);
           return h.response(await response.text()).type('text/plain');
       }
     });
 
+    server.route({
+        method: 'GET',
+        path: '/diffs/{id}',
+        handler: async (request, h) => {
+            const { id } = request.params;
+
+            const response = await pool.query(
+                'SELECT * FROM diffs WHERE id = $1',
```

Comment:
I wonder if this could already do the JSON parsing, with something like `FOR JSON PATH`, which I found here: https://blog.jooq.org/2019/11/13/stop-mapping-stuff-in-your-middleware-use-sqls-xml-or-json-operators-instead/